x-pack/solutions/observability/plugins/apm/server/utils/get_field_value.ts

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export const getFieldValue = <T>(
```

Comment:
q: Why do we need this function?

There is already a way to get field values from Elasticsearch responses in APM using `unflattenKnownApmEventFields`. My concern is that introducing another option for doing the same thing could lead to inconsistencies.

If we decide that this function is better, I’d prefer that it comes with a plan to gradually replace `unflattenKnownApmEventFields,` or that we explore ways to address any existing issues in `unflattenKnownApmEventFields`.

Reply:
`unflattenKnownApmEventFields` is a very heavy function that turns the processing of TraceFields into an O(n^2) process in both iteration and allocations, and we have known APM fields already as string constants, so we don't actually need to impose such overhead, which becomes particularly noticeable on large traces. I actually want to nominate it for removal precisely because we shouldn't be using utils that cause our data processing to becoming O(n^2), given we already have problems with performance on various API calls.